### PR TITLE
fix: improve downgrade flow in onboarding

### PR DESCRIPTION
**Pull request by raquelmsmith**
## Problem

If someone upgrades in onboarding, then decides to downgrade, they get an error.

## Changes

- Fixes the downgrade link so that it doesn't point to an activation URL.
- Handles downgrade action in the button
- Adds loading states for both upgrade and downgrade

https://github.com/PostHog/posthog/assets/18598166/012441a5-e09a-409b-b2b0-6629149e46fb

After downgrade it flickers saying "success" again for a split second... I don't think that many people will use this so I'm not too concerned about it.

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

Not available on self-hosted

## How did you test this code?

👀 🖱️

### CODE REVIEW

File: frontend/src/scenes/billing/billingProductLogic.ts [hunk 1]
```diff
@@ -30,6 +30,7 @@ export const billingProductLogic = kea<billingProductLogicType>([
             [
                 'updateBillingLimits',
                 'updateBillingLimitsSuccess',
+                'loadBilling',
```
Comment (zlwaterfield) [inline frontend/src/scenes/billing/billingProductLogic.ts:33]: Are you using this?

File: frontend/src/scenes/billing/billingProductLogic.ts [hunk 2]
```diff
@@ -266,6 +267,7 @@ export const billingProductLogic = kea<billingProductLogicType>([
                     ? actions.reportSurveySent(values.surveyID, values.surveyResponse)
                     : actions.reportSurveyDismissed(values.surveyID)
             }
+            location.reload()
```
Comment (zlwaterfield) [inline frontend/src/scenes/billing/billingProductLogic.ts:270]: Why is the reload needed?

**Review by zlwaterfield — APPROVED** [review]
just a few questions but looks good

File: frontend/src/scenes/billing/billingProductLogic.ts (hunk 1 above)
Comment (raquelmsmith) [reply frontend/src/scenes/billing/billingProductLogic.ts]: nope! fixed

File: frontend/src/scenes/onboarding/OnboardingBillingStep.tsx [hunk 3]
```diff
@@ -57,7 +58,7 @@ export const OnboardingBillingStep = ({
                 )
             }
         >
-            {billing?.products && productKey && product ? (
+            {billing?.products && productKey && product && !billingLoading ? (
```
Comment (xrdt) [inline frontend/src/scenes/onboarding/OnboardingBillingStep.tsx:61]: Would it be easier to have billingloading be a top level check? 

`{billingLoading ? <Spinner> : .... }`

File: frontend/src/scenes/billing/billingLogic.tsx [hunk 4]
```diff
@@ -172,6 +172,12 @@ export const billingLogic = kea<billingLogicType>([
                 },
 
                 deactivateProduct: async (key: string) => {
+                    // clear upgrade params from URL
+                    const currentURL = new URL(window.location.href)
+                    currentURL.searchParams.delete('upgraded')
+                    currentURL.searchParams.delete('products')
```
Comment (xrdt) [inline frontend/src/scenes/billing/billingLogic.tsx:178]: Is this necessary? The page will reload upon deactivateProduct success.

**Review by xrdt — APPROVED** [review]

File: frontend/src/scenes/billing/billingLogic.tsx (hunk 4 above)
Comment (raquelmsmith) [reply frontend/src/scenes/billing/billingLogic.tsx:178]: Yeah, if those are there the page thinks an upgrade happened successfully, so we need to make sure they're removed before reload.

**Comment by raquelmsmith** [comment]
I'm going to pick this back up after subscribe-to-all-products is out

**Comment by zlwaterfield** [comment]
I've updated this branch to support the change from subscribe to all products. I've also made a change to improve loading. Previously, when there was no current step, it showed a blank screen; now, it will show a loader.

File: frontend/__snapshots__/scenes-app-dashboards--insight-legend--light.png
Comment (zlwaterfield) [inline frontend/__snapshots__/scenes-app-dashboards--insight-legend--light.png:1]: this is flakey, pr didn't touch anything related

**Review by raquelmsmith — COMMENTED** [review]
Looks good, works good! ✅

**Comment by raquelmsmith** [comment]
Cypress test failure looks legit https://github.com/PostHog/posthog/actions/runs/10072575554/job/27844845724?pr=22436

**Comment by zlwaterfield** [comment]
> Cypress test failure looks legit [PostHog/posthog/actions/runs/10072575554/job/27844845724?pr=22436](https://github.com/PostHog/posthog/actions/runs/10072575554/job/27844845724?pr=22436)

will look!

**Pull request merged**